hack/externalnode/install-vm.sh

```diff
@@ -0,0 +1,174 @@
+#!/usr/bin/env bash
+
+# Copyright 2022 Antrea Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -eo pipefail
+
+function echoerr {
+    >&2 echo "$@"
+}
+
+_usage="Usage: $0 [--ns <NameSpace>] [--bin <AntreaAgentSavePath>] [--config <AgentConfigSavePath>] [--kubeconfig <KubeconfigSavePath>] [--antrea-kubeconfig <AntreaKubeconfigSavePath>] [--nodename <ExternalNodeName>] [--help|-h]
+        --ns                          NameSpace to be used by the antrea-agent.
+        --bin                         Path of the antrea-agent binary
+        --config                      Path of the antrea-agent configuration file
+        --kubeconfig                  Path of the kubeconfig to access K8s API Server
+        --antrea-kubeconfig           Path of the kubeconfig to access Antrea API Server
+        --nodename                    ExternalNode name to be used by the antrea-agent
+        --help, -h                    Print this message and exit"
+
+function print_usage {
+    echoerr "$_usage"
+}
+
+function print_help {
+    echoerr "Try '$0 --help' for more information."
+}
+
+INSTALL_PATH="/usr/sbin"
+AGENT_BIN_PATH=""
+CONFIG_PATH=""
+KUBECONFIG=""
+ANTREAKUBECONFIG=""
+AGENT_NAMESPACE=""
+NODE_NAME="$(hostname)"
+AGENT_LOG_DIR="/var/log/antrea"
+AGENT_CONF_PATH="/etc/antrea"
+
+check_supported_platform() {
+  echo "Checking platform supported"
+  if ! [[ $(lsb_release -rs) =~ ^(18.04|20.04)$ ]]; then
+    echoerr "Error only Ubuntu 18.04/20.04 version is supported"
+    exit 1
+  fi
+}
+
+copy_antrea_agent_files() {
```

Comment:
One question: it seems we require the user to manually updates the configuration file first, and then the script only generates the kubeconfig files and appends them into the configuration (also updates the namespace). But for other configuration items, they depend on the users manual modifcation. 

I originally thought the idea is, user only executes the scripts, and provides the necessary config values, like namespace, policyBypassRules, etc, and the script downloads the sample configuration file directly, and generates the final configuration file in one step. 

This question is suitable for both Linux and Windows

@reachjainrahul Would you share your thought?

Reply:
Having offline discussion with @reachjainrahul , the conclusion is we could download antrea-agent bits/configuration file in a future patch, since the download url for these files are not provided before release.